app/models/assignment_repo.rb

```diff
@@ -6,14 +6,22 @@ class AssignmentRepo < ActiveRecord::Base
 
   belongs_to :assignment
   belongs_to :repo_access
+  belongs_to :user
 
   validates :assignment, presence: true
 
   validates :github_repo_id, presence:   true
   validates :github_repo_id, uniqueness: true
 
-  validates :repo_access, presence:   true
-  validates :repo_access, uniqueness: { scope: :assignment }
+  before_validation(on: :create) do
+    if organization
+      create_github_repository
+      push_starter_code
+      add_user_as_collaborator
```

Comment:
If `create_github_repository` succeeds but `push_starter_code` fails, we need to remove the repository from GitHub. I think this is handled correctly here since we're getting these methods from the `GitHubRepoable` module, right?